### ISSUE TITLE
Capture exit code when Vite process is killed

### ISSUE THREAD

**Issue by JasonBarnabe** [ElMassimo/vite_ruby#292]
[troubleshooting section]: https://vite-ruby.netlify.app/guide/troubleshooting.html

- [ ] I have tried upgrading by running `bundle update vite_ruby`.
- [x] I have read the __[troubleshooting section]__ before opening an issue.

### Description 📖

Running `bin/rails assets:precompile --trace` on CircleCI, my build ends with:

```
Use of eval is strongly discouraged, as it poses security risks and may cause issues with minification
Use of eval is strongly discouraged, as it poses security risks and may cause issues with minification
Use of eval is strongly discouraged, as it poses security risks and may cause issues with minification
Use of eval is strongly discouraged, as it poses security risks and may cause issues with minification

Build with Vite failed! ❌
```

With no further message as to why. Adding some debugging to `lib/vite_ruby/io.rb` reveals that when `vite build` is run, `stderr` contains only the warning messages, but `wait_threads.value` is `pid 11560 SIGKILL (signal 9)`, so presumably something on CircleCI is killing it (OOM killer?).

vite_ruby should output the return code of the `vite build` command in addition to `stderr` when the build fails to make it easier to diagnose the problem.

### Reproduction 🐞

Run `bin/rails assets:precompile --trace`, when it gets to the vite part, `kill -9` the vite process. Result:

```
Build with Vite failed! ❌
```
With no explanation why.

<details>
<summary>Vite Ruby Info</summary>
bin/vite present?: true
vite_ruby: 3.2.6
vite_rails: 3.0.12
rails: 6.1.6
node: v16.18.0
npm: 8.5.0
yarn: 1.22.5
pnpm: 
ruby: ruby 2.7.6p219 (2022-04-12 revision c9c2245c0a) [x86_64-linux]

installed packages:
project@0.0.20 /home/circleci/project
├─┬ @vitejs/plugin-basic-ssl@0.1.2
│ └── vite@3.1.8 deduped
├─┬ @vitejs/plugin-react@2.2.0
│ └── vite@3.1.8 deduped
├─┬ vite-plugin-relay@2.0.0
│ └── vite@3.1.8 deduped
├─┬ vite-plugin-ruby@3.1.2
│ └── vite@3.1.8 deduped
└── vite@3.1.8
</details>

**Comment by ElMassimo** [comment]
Pull requests are welcome!

**Comment by ElMassimo** [comment]
Thanks for your contribution, released in `vite_ruby-3.2.9`.
